provider-assignments/src/java/org/sakaiproject/archiver/provider/AssignmentArchiver.java

```diff
@@ -0,0 +1,408 @@
+	private ContentHostingService contentHostingService;
+
+	@Setter
+	private AssignmentService assignmentService;
+
+	@Setter
+	private ArchiverService archiverService;
+	
+	@Setter
+	private SiteService siteService;
+	
+	@Setter
+	private UserDirectoryService userDirectoryService;
+
+	@Setter
+	private AssignmentSupplementItemService assignmentSupplementItemService;
+
+	@Override
+	public void archive(final String archiveId, final String siteId, final String toolId, final boolean includeStudentContent) {
+		
+		List<Assignment> assignments = this.assignmentService.getListAssignmentsForContext(siteId);
+
+		for (Assignment assignment : assignments) {
+
+			// archive the assignment data
+			SimpleAssignment simpleAssignment = new SimpleAssignment(assignment);
+			this.archiverService.archiveContent(archiveId, siteId, toolId, Jsonifier.toJson(simpleAssignment).getBytes(), "details.json", assignment.getTitle());
+
+			// archive the attachments for the assignment
+			for (Reference attachment : assignment.getContent().getAttachments()) {
+				try {
+					archiveAttachments(attachment, archiveId, siteId, toolId, assignment.getTitle() + "/attachments");
+				} catch (ServerOverloadException | IdUnusedException | TypeException | PermissionException e) {
+					log.error("Error getting attachment for assignment: " + assignment.getTitle());
+					continue;
+				} 
+			}
+			
+			// archive the grades spreadsheet for the assignment
+//			byte[] gradesSpreadsheet;
+//			try {
+//				gradesSpreadsheet = this.assignmentService.getGradesSpreadsheet(this.assignmentService.gradesSpreadsheetReference(siteId, assignment.getId()));
+//				if (gradesSpreadsheet != null ) {
+//					this.archiverService.archiveContent(archiveId, siteId, toolId, gradesSpreadsheet, "grades.csv", assignment.getTitle());
+//				}
+//			} catch (IdUnusedException | PermissionException e) {
+//				log.error("Error getting grades spreadsheet for assignment: " + assignment.getTitle());
+//				continue;
+//			}
+			
+			// if we want student content, archive the submissions for the assignment
+			if (includeStudentContent) {
+				this.archiveSubmissions(assignment, archiveId, siteId, toolId);
+			}
+		}
+	}
+
+	/**
+	 * Get the submissions for an assignment, and any feedback from the instructor
+	 * @param assignment
+	 * @param archiveId
+	 * @param siteId
+	 * @param toolId
+	 */
+	@SuppressWarnings("unchecked")
+	private void archiveSubmissions(Assignment assignment, String archiveId, String siteId, String toolId) {
+		List<AssignmentSubmission> submissions = this.assignmentService.getSubmissions(assignment);
+		
+		for (AssignmentSubmission submission : submissions) {
+			
+			// get the attachments for this submission
+			for (Object submissionObj : submission.getSubmittedAttachments()) {
+				String subdirs = getSubDirs(assignment, submission.getSubmitterId(), "submission", siteId);
+				
+				try {
+					archiveAttachments((Reference)submissionObj, archiveId, siteId, toolId, subdirs);
```

Comment:
add additional subdir here so that every user submission is inside a 'submissions' dir in the archive